Source/WebKit/UIProcess/API/wpe/WebKitWebViewWPE.cpp

```diff
@@ -30,9 +30,18 @@
 #include <wpe/wpe-platform.h>
 #endif
 
-gboolean webkitWebViewAuthenticate(WebKitWebView*, WebKitAuthenticationRequest*)
+gboolean webkitWebViewAuthenticate(WebKitWebView*, WebKitAuthenticationRequest* request)
 {
-    return FALSE;
+    switch (webkit_authentication_request_get_scheme(request)) {
+    case WEBKIT_AUTHENTICATION_SCHEME_CLIENT_CERTIFICATE_REQUESTED:
+        webkit_authentication_request_authenticate(request, nullptr);
+        break;
+    default:
+        webkit_authentication_request_cancel(request);
```

Comment:
hmm, that probably means the request is leaked, because it's cancelled in its destructor. This will also cause the load to fail, but I guess we don't want to fail, but continue and show the unauthorized page, no? So, I think we always want to call authenticate with nullptr for all schemes.

Reply:
The destructor cancels too, but only the first `webkit_authentication_request_cancel` call will do something.
The request would be freed when the reference is released in `webkitWebViewCompleteAuthenticationRequest`, but this function is focused on the main resource, which is strange. Maybe authentication requests done for resources would be temporarily leaked until the next navigation, or next request.

The cancel button of the Gtk authentication dialog calls authenticate with nullptr, so you're right, it's probably the best option.